backend/src/sync/harvest/mod.rs

```diff
@@ -473,3 +481,53 @@ async fn upsert(
     let statement = db.prepare_cached(&*query).await?;
     Ok(db.query_one(&statement, &values).await?.get::<_, i64>(0))
 }
+
+
+/// Removes realms that reference a deleted series, event, or playlist,
+/// if the configuration allows it.
+///
+/// If realm deletion is enabled for the given type, the function:
+/// 1. Finds all realms whose names are derived from the deleted series/event/playlist.
+/// 2. Ensures that those realms have no child realms.
+/// 3. If `eager = false`, ensures that the realm has **only one** block (the deleted one).
+/// 4. Deletes the qualifying realms from the database.
+async fn remove_realms(
+    db: &deadpool_postgres::Transaction<'_>,
+    config: &Config,
+    item: &HarvestItem,
+) -> Result<(), tokio_postgres::Error> {
+    let Some(props) = item.deleted_props(config) else {
+        return Ok(());
+    };
+
+    let block_count_condition = if props.eager {
+        String::new()
+    } else {
+        "and (select count(*) from blocks b2 where b2.realm = r.id) = 1".to_owned()
+    };
+
+    let query = format!(" \
+        delete from realms r \
+        where r.name_from_block in ( \
+            select b.id from blocks b \
+            join {table_name} t on b.{block_type} = t.id \
+            where t.opencast_id = $1 and b.type = '{block_type}' \
+        ) \
+        and not exists ( \
+            select 1 from realms child where child.parent = r.id \
+        ) \
+        {block_count_condition}",
+        table_name = props.table_name,
+        block_type = props.block_type.to_string(),
+    );
+
+    let rows_affected = db.execute(&query, &[&props.id]).await?;
+    info!(
+        "Removed {rows_affected} realms referencing {} {} (eager={})",
+        props.block_type,
+        props.id,
+        props.eager
+    );
```

Comment:
Remember that our log macros have special features to pass in variables. Sth like this?

```rust
info!(
    eager,
    "Removed {rows_affected} realms referencing {} {}",
    props.block_type,
    props.id,
);
```

Some things are nicer to be interpolated into a human sentence, but the `eager = {}` thing would be a clear candidate to use these log "fields".